src/main/java/org/jabref/logic/integrity/YearChecker.java

```diff
@@ -35,6 +40,14 @@
             return Optional.of(Localization.lang("last four nonpunctuation characters should be numerals"));
         }
 
+        if(!Character.isDigit(C_value[0])){
```

Comment:
Do you use Eclipse or IntelliJ? In IntelliJ, we have custom format rules: https://github.com/JabRef/jabref/wiki/Guidelines-for-setting-up-a-local-workspace#intellij. Maybe, we generate the Eclipse rules also at `gradlew eclipse` (see https://github.com/JabRef/jabref/wiki/Guidelines-for-setting-up-a-local-workspace#set-up-eclipse)